wfapi/error.go

```diff
@@ -137,28 +71,27 @@ func ErrorSearchingFilesystem(searchingFor string, cause error) Error {
 // Errors:
 //
 //  - warpforge-error-invalid --
-func ErrorInvalid(message string, deets ...[2]string) Error {
-	return &ErrorVal{
-		CodeString: "warpforge-error-invalid",
-		Message:    message,
-		Details:    deets,
+func ErrorInvalid(message string, deets ...[2]string) error {
```

Comment:
I think for some of these we maybe can just not bother having special constructor functions at all.  I think it's sorta fine if we see the serum package's constructors being used across the codebase.

The biggest reason we had a constructor for every single error here before was that we had really no other syntactically reasonable way to avoid it.  Now I think using the serum package widely is fine.  Additional constructor functions in this package are now only useful if they can take structured parameters (either well-typed, or a series of things that should always be specified together and we want to use the function to remind us).

This line I'm commenting on is a particularly outstanding not-worth-it-anymore, but there's probably others that don't make the cut anymore too.

Reply:
Seems reasonable. I'm not super prepped to be making those decisions tonight though.

Reply:
Right, I...wish I could deploy a small time machine to save you from already having written this bit, heh.

Reply:
Honestly nothing stands out to me as egregiously bad.